### PR TITLE
Add tap-mongodb variant menzenski

### PR DESCRIPTION
**Pull request by menzenski**
## Checklist

- [X] Add/update the file in the appropriate folder (`/taps` or `/targets`). The name of the file should match the name of the tap. If there is already one, add a descriptor to the name such as `-search`.
- [X] Add/update the PNG logo image in `/assets/logos/<taps or targets>`. The image name must match the YAML file name.
- [X] Tag `@tayloramurphy` or `@pnadolny13` to flag it for review. Or post to the [#hub](https://meltano.slack.com/archives/C01UGBSJNG5) channel on Meltano slack.

## Reviewer Checklist

- [ ] Validate file against JSON Schema. https://www.jsonschema.net/home is an option.
- [ ] Build website locally and validate everything works.

### CODE REVIEW

File: _data/meltano/extractors/tap-mongodb/menzenski.yml [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+logo_url: /assets/logos/extractors/mongodb.png
+maintenance_status: beta
+name: tap-mongodb
+namespace: tap_mongodb
+next_steps: ''
+pip_url: git+https://github.com/menzenski/tap-mongodb.git
+repo: https://github.com/menzenski/tap-mongodb
+settings:
+- description: Database name from which records will be extracted.
+  kind: string
```
Comment (pnadolny13) [inline _data/meltano/extractors/tap-mongodb/menzenski.yml:23]:
```suggestion
  kind: string
  label: Database
```

File: _data/meltano/extractors/tap-mongodb/menzenski.yml [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+  name: database
+- description: |
+    MongoDB connection string. See
+    https://www.mongodb.com/docs/manual/reference/connection-string/#connection-string-uri-format for specification.
+    The password included in this string should be url-encoded. The tap will not url-encode it.
+  kind: password
```
Comment (pnadolny13) [inline _data/meltano/extractors/tap-mongodb/menzenski.yml:29]:
```suggestion
  kind: password
  label: MongoDB Connection String
```

File: _data/meltano/extractors/tap-mongodb/menzenski.yml [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+  name: mongodb_connection_string
+- description: |
+    String (serialized JSON object) with keys 'username', 'password', 'engine', 'host', 'port',
+    'dbClusterIdentifier' or 'dbName', 'ssl'. See example at
+    https://docs.aws.amazon.com/secretsmanager/latest/userguide/reference_secret_json_structure.html#reference_secret_json_structure_docdb.
+    The password from this JSON object will be url-encoded by the tap before opening the database connection.
+    (This config setting exists to support use of AWS SecretsManager to manage a MongoDB/DocumentDB credenial).
+  kind: password
```
Comment (pnadolny13) [inline _data/meltano/extractors/tap-mongodb/menzenski.yml:37]:
```suggestion
  kind: password
  label: DocumentDB Credential JSON String
```

File: _data/meltano/extractors/tap-mongodb/menzenski.yml [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+  name: documentdb_credential_json_string
+- description: |
+    String (serialized JSON object) containing string-string key-value pairs which will be added to the
+    connection string options when using documentdb_credential_json_string. For example, when set to
+    the string `{\"tls\":\"true\",\"tlsCAFile\":\"my-ca-bundle.pem\"}`, the options
+    `tls=true&tlsCAFile=my-ca-bundle.pem` will be passed to the MongoClient.
+  kind: string
```
Comment (pnadolny13) [inline _data/meltano/extractors/tap-mongodb/menzenski.yml:44]:
```suggestion
  kind: string
  label: DocumentDB Credential JSON Extra Options
```

File: _data/meltano/extractors/tap-mongodb/menzenski.yml [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+  name: documentdb_credential_json_extra_options
+- description: |
+    An optional prefix which will be added to each stream name.
+  kind: string
```
Comment (pnadolny13) [inline _data/meltano/extractors/tap-mongodb/menzenski.yml:48]:
```suggestion
  kind: string
  label: Prefix
```

File: _data/meltano/extractors/tap-mongodb/menzenski.yml [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+  name: prefix
+  value: ''
+- description: |
+    Start date - used for incremental replication only. Log based replication does not support this setting - do
+    not provide it unless you are using incremental replication. Defaults to epoch zero time `1970-01-01` if
+    tap uses incremental replication method.
+  kind: date_iso8601
```
Comment (pnadolny13) [inline _data/meltano/extractors/tap-mongodb/menzenski.yml:55]:
```suggestion
  kind: date_iso8601
  label: Start Date
```

File: _data/meltano/extractors/tap-mongodb/menzenski.yml [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+  name: start_date
+  value: '1970-01-01'
+- description: |
+    When True, _sdc metadata fields will be added to records produced by this tap. If the tap is run in log-based
+    replication mode, if this setting is enabled, the `_sdc_extracted_at` and `_sdc_deleted_at` timestamps on records
+    will be set to the cluster time value from the database change stream event.
+  kind: boolean
```
Comment (pnadolny13) [inline _data/meltano/extractors/tap-mongodb/menzenski.yml:62]:
```suggestion
  kind: boolean
  label: Add Record Metadata
```

File: _data/meltano/extractors/tap-mongodb/menzenski.yml [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+- description: |
+    In AWS DocumentDB (unlike MongoDB), change streams must be enabled specifically (see
+    https://docs.aws.amazon.com/documentdb/latest/developerguide/change_streams.html#change_streams-enabling
+    ). If attempting to open a change stream against a collection on which change streams have not been
+    enabled, an OperationFailure error will be raised. If this property is set to True, when this error
+    is seen, the tap will execute an admin command to enable change streams and then retry the read
+    operation. Note: if this setting is enabled, the credential the tap is using must have the `modifyChangeStreams`
+    permission. Second note: use of this setting, and of change streams in general, may incur additional costs in
+    AWS DocumentDB.
+  kind: boolean
```
Comment (pnadolny13) [inline _data/meltano/extractors/tap-mongodb/menzenski.yml:73]:
```suggestion
  kind: boolean
  label: Allow Modify Change Streams
```

File: _data/meltano/extractors/tap-mongodb/menzenski.yml [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+  name: allow_modify_change_streams
+- description: |
+    List of MongoDB change stream operation types to include in tap output. The default behavior is to
+    limit to document-level operation types. See full list of operation types at
+    https://www.mongodb.com/docs/manual/reference/change-events/#operation-types. Note that the list
+    of allowed_values for this property includes some values not available to all MongoDB versions.
+  kind: array
```
Comment (pnadolny13) [inline _data/meltano/extractors/tap-mongodb/menzenski.yml:80]:
```suggestion
  kind: array
  label: Operation Types
```

**Review by pnadolny13 — CHANGES_REQUESTED** [review]
@menzenski this is awesome! Thanks for adding the detailed settings preamble so users can understand why this variant is different than others. I just left some suggestions to add labels so the hub rendering is prettier.

**Comment by menzenski** [comment]
> @menzenski this is awesome! Thanks for adding the detailed settings preamble so users can understand why this variant is different than others. I just left some suggestions to add labels so the hub rendering is prettier.

Thanks @pnadolny13 - I've accepted all those label additions.

**Review by pnadolny13 — APPROVED** [review]
🚀

**Pull request merged**